File: stack/150-Evaluate-Reverse-Polish-Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if token in {\+\, \-\, \*\, \/\}:
                a = stack.pop()
                b = stack.pop()
                if token == \+\:
                    stack.append(b + a)
                elif token == \-\:
                    stack.append(b - a)
                elif token == \*\:
                    stack.append(b * a)
                elif token == \/\:
                    # Use integer division and truncate towards zero
                    stack.append(int(b / a))
            else:
                stack.append(int(token))
        
        return stack[-1]

File: stack/150-Evaluate-Reverse-Polish-Notation.go
func evalRPN(tokens []string) int {
    if len(tokens) <= 1 {
        value, _ := strconv.Atoi(tokens[0])
        return value
    }

    var stacks []int
    for _, token := range tokens {
        switch token {
            case \+\:
                firstNumber := stacks[len(stacks) - 1]
                secondNumber := stacks[len(stacks) - 2]
                stacks = stacks[:len(stacks) - 2]
                stacks = append(stacks, secondNumber + firstNumber)
            case \-\:
                firstNumber := stacks[len(stacks) - 1]
                secondNumber := stacks[len(stacks) - 2]
                stacks = stacks[:len(stacks) - 2]
                stacks = append(stacks, secondNumber - firstNumber)
            case \*\:
                firstNumber := stacks[len(stacks) - 1]
                secondNumber := stacks[len(stacks) - 2]
                stacks = stacks[:len(stacks) - 2]
                stacks = append(stacks, secondNumber * firstNumber)
            case \/\:
                firstNumber := stacks[len(stacks) - 1]
                secondNumber := stacks[len(stacks) - 2]
                stacks = stacks[:len(stacks) - 2]
                stacks = append(stacks, secondNumber / firstNumber)
            default:
                number, _ := strconv.Atoi(token)
                stacks = append(stacks, number)
        }
    }

    return stacks[len(stacks) - 1]
}